src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpHeaders.cs

```diff
@@ -159,28 +153,38 @@ internal bool TryAddWithoutValidation(HeaderDescriptor descriptor, IEnumerable<s
                 throw new ArgumentNullException(nameof(values));
             }
 
-            using (IEnumerator<string?> enumerator = values.GetEnumerator())
+            using IEnumerator<string?> enumerator = values.GetEnumerator();
+            if (enumerator.MoveNext())
             {
+                string firstValue = enumerator.Current ?? string.Empty;
                 if (enumerator.MoveNext())
                 {
-                    TryAddWithoutValidation(descriptor, enumerator.Current);
-                    if (enumerator.MoveNext())
+                    var valuesList = new List<string>
                     {
-                        HeaderStoreItemInfo info = GetOrCreateHeaderInfo(descriptor, parseRawValues: false);
-                        do
-                        {
-                            AddRawValue(info, enumerator.Current ?? string.Empty);
-                        }
-                        while (enumerator.MoveNext());
+                        firstValue
+                    };
+
+                    do
+                    {
+                        valuesList.Add(enumerator.Current ?? string.Empty);
```

Comment:
Interesting that we add empty strings in the old code. I don't think they are semantically meaningful.